### PR TITLE
Disable implicit URL normalization

### PR DESCRIPTION
**Pull request by vitalca**
This patch fixes retrieving of files with "+" in the name

### CODE REVIEW

**Comment by vitalca** [comment]
This is to fix issue #144

**Comment by uwolfer** [comment]
Thank you for this PR. At a first glance it looks good, but I'm still a bit scarred of breaking things here because the impact could be quite big. Have you found any documentation or code in Gerrit which explains why we should do it this way?

**Comment by vitalca** [comment]
> Have you found any documentation or code in Gerrit which explains why we should do it this way?

Looks like it has nothing to do with Gerrit and your code. It is an Apache HttpClient problem. Specifically, its default configuration. I don't think there is any other reasonable way to fix it.

I described everything in #144. If you follow the code step-by-step, you'll see that Gerrit Client passes the correctly encoded URL to HttpClient which later decodes it back because `RequestConfig' has not been set and therefore it uses the default value, which causes the problem.

In short, my task was to figure out a way to disable this unwanted behavior of Apache HttpClient.

**Comment by uwolfer** [comment]
@vitalca: Thank you very much for your detailed analysis and the proposed fix. Looks good to me, but I will do some further tests before merging.

In another project which depends on this lib, there is a "workaround" implemented which is might related to this fix: https://github.com/uwolfer/gerrit-intellij-plugin/blob/4e3aa3d907c2e927108bc4c87de5cbb9943df326/src/main/java/com/urswolfer/intellij/plugin/gerrit/ui/filter/FulltextFilter.java#L65-L76
What do you think? Can that be removed once this fix is in place?

**Comment by vitalca** [comment]
Hi @uwolfer,
Honestly I don't find much point in the `specialEncodeFulltextQuery()` you are referring to. The `URLEncoder.encode()` is implemented in accordance to the [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986) requirements, section 2.2 (an alternative [link to Wikipedia](https://en.wikipedia.org/wiki/Percent-encoding#Reserved_characters)). The characters you are trying to ignore actually must be escaped. But perhaps I am missing some background here... 

If you really need to keep this custom encoding logic, then my fix won't change much, because It does the opposite: it disables the implicit unnecessary URL **de**coding in the Apache HttpClient, added in v.4.5.8.

Anyway, FYI my small fix has been in production for 2 weeks now. Nobody has complained so far. I had to deploy it ASAP because of a high-priority ticket on my side.

**Comment by uwolfer** [comment]
@vitalca: Thanks for your explanation.

@reda-alaoui: I'm about to merge this PR. Would you like to test it in your setup before?

**Comment by reda-alaoui** [comment]
@uwolfer Thank you for the notification. I just tested the change on my setup and it works fine :+1:

**Comment by uwolfer** [comment]
@vitalca Thanks for your work!

**Comment by vitalca** [comment]
My pleasure :)
